source/common/formatter/substitution_formatter.h

```diff
@@ -113,14 +113,22 @@ class JsonFormatterImpl : public Formatter {
                      absl::string_view local_reply_body) const override;
 
 private:
+  struct JsonFormatMap;
+  using JsonFormatMapValue =
+      absl::variant<const std::vector<FormatterProviderPtr>, const JsonFormatMap>;
+  struct JsonFormatMap {
+    std::unique_ptr<std::map<std::string, JsonFormatMapValue>> value_;
```

Comment:
We usually have a typedef for this. But I don't feel strongly about it.

Reply:
Added some aliases. Looks more readable indeed now.